proxy/proxy.go

```diff
@@ -1458,6 +1458,7 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	defer func() {
 		if ctx.response != nil && ctx.response.Body != nil {
+			ctx.responseWriter.Flush()
```

Comment:
I think there will be no context response in case of net error (that is handled by sendError), might worth to have separate unconditional defer...

Reply:
done